### PR TITLE
[ENT-432] Remove EnterpriseIntegratedChannel model.

### PR DESCRIPTION
**Pull request by UmanShahzad**
The `EnterpriseIntegratedChannel` model was unused and it seems unnecessary.

**JIRA:** [ENT-432](https://openedx.atlassian.net/browse/ENT-432)

**Merge deadline:** End of sprint V.

**Testing instructions:**

This code was unused, so this should not change any functionality.

1. Make sure the migration occurs correctly.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @UmanShahzad! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=224&repo=edx%2Fedx-enterprise"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by UmanShahzad** [comment]
Note that this won't require a special/careful migration because the feature is unused.

**Comment by UmanShahzad** [comment]
@douglashall I realized there were a few more things to delete that were totally unused -- 80c8cde44a08d6f86ae97a4fbb6378a972e1a362.

File: enterprise/conf/locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -577,14 +577,95 @@ msgid ""
...
+msgstr ""
+
+#: enterprise/models.py:1226
+msgid "Day of month must be set if the frequency is monthly."
+msgstr ""
+
+#: enterprise/models.py:1229
+msgid "Frequency must be set to either daily, weekly, or monthly."
+msgstr ""
+
```
Comment (itsjeyd) [inline enterprise/conf/locale/en/LC_MESSAGES/django.po]: @UmanShahzad This PR is code removal only (aside from the migration that removes model data), so why add any contents here? :)

Comment (UmanShahzad) [reply enterprise/conf/locale/en/LC_MESSAGES/django.po]: I didn't -- `make extract_translations` did. I think someone just forgot to run it previously.

**Comment by UmanShahzad** [comment]
Rebased on master from `80c8cde44a08d6f86ae97a4fbb6378a972e1a362`

File: enterprise/conf/locale/en/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -1215,19 +1215,3 @@ msgstr ""
...
-msgid "Data Type"
-msgstr ""
-
-#: integrated_channels/integrated_channel/models.py:35
-msgid "Enterprise Integrated Channel"
-msgstr ""
-
-#: integrated_channels/integrated_channel/models.py:36
-msgid "Enterprise Integrated Channels"
-msgstr ""
```
Comment (UmanShahzad) [inline enterprise/conf/locale/en/LC_MESSAGES/django.po]: So looks like someone already got to fixing the missing translations before me, @itsjeyd, so those additions are gone.

**Comment by UmanShahzad** [comment]
Codecov is failing because after rebasing it found that we went from 100% to less since `master` was missing some coverage from a separate PR. The code in question: https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L980-L981

So I'm going to ignore that for now -- @zubair-arbi can you or someone handle this coverage in a PR that could be related?

**Pull request merged**